### DISCUSSION TITLE
rpc call with json params raises an sql error: identifier will be truncated to ....

### DISCUSSION THREAD

**Discussion by fusionbeam** [PostgREST/postgrest#1910] [Q&A] [answered]
### Environment

* PostgreSQL version: 13.3 (Ubuntu 13.3-1.pgdg20.04+)
* PostgREST version: 7.0.1
* Operating system: Ubuntu 20.04.1

### Description of issue

Calling an RPC with 3 json parameters causes an SQL error: 
``
NOTICE:  identifier "{"brand_details":{"app_id":"asht","name":"test news"},"owner_details":{"app_id":"asht","name":"test news","active_profiles":["ashta_brand_owner","ashta_distributor"]},"owner_services":[{"app_id":"asht","name":"test news","naics_code":"3159","provisionings":{"contract_type": "distributor","retailer_connection_approval": "any","retailer_connection_fee_bearer": "brand"}},{"app_id":"asht","name":"test news","naics_code":"4243","provisionings":{"contract_type": "distributor","retailer_connection_approval": "any","retailer_connection_fee_bearer": "distributor"}}]}" will be truncated to "{"brand_details":{"app_id":"asht","name":"test news"},"owner_de" 
``

Any hints onto what may be problem?
I've reloaded the schema, restarted everything, the function name is correct, parameter names are correct ...

here are the exact parameters and their value that are causing this error.
``
curl -X POST "https://api....../rpc/board_new_brand" -H  "accept: application/json"  -d '{"brand_details":{"app_id":"asht","name":"test news"},"owner_details":{"app_id":"asht","name":"test news","active_profiles":["ashta_brand_owner","ashta_distributor"]},"owner_services":[{"app_id":"asht","name":"test news","naics_code":"3159","provisionings":{"contract_type": "distributor","retailer_connection_approval": "any","retailer_connection_fee_bearer": "brand"}},{"app_id":"asht","name":"test news","naics_code":"4243","provisionings":{"contract_type": "distributor","retailer_connection_approval": "any","retailer_connection_fee_bearer": "distributor"}}]}'
``

I can run this command against any schema endpoint it causes the same error.

**Comment by steve-chavez** [comment, marked as answer]
> Calling an RPC with 3 json parameters causes an SQL error:

It looks like a json key+value is being sent as a function parameter name and that surpasses the postgres identifier [max length](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS)(63 chars), that's why the truncation happens.

> curl -X POST "https://api....../rpc/board_new_brand" -H "accept: application/json"

Try changing `-H "accept: application/json"` to `-H "Content-Type: application/json"`.

**Comment by fusionbeam** [comment]
Life saver! Thank you.
